tests/unit/steps/test_steps_plan.py

```diff
@@ -34,30 +34,25 @@ async def test_generate_plan(mocker, apps):
     assert plan.description == "Top level plan"
     assert not plan.parallel
     assert not plan.function
-    assert len(plan.sub_steps) == 2
+    assert len(plan.sub_steps) == 3
 
     sub_step_back_up = plan.sub_steps[0]
     assert sub_step_back_up.description == "backup mysql databases"
     assert not sub_step_back_up.parallel
     assert sub_step_back_up.function == backup
 
     sub_step_upgrade_plan = plan.sub_steps[1]
-    assert sub_step_upgrade_plan.description == "Application(s) upgrade plan"
+    assert sub_step_upgrade_plan.description == "Principal(s) upgrade plan"
```

Comment:
I think it would be good to include `keystone-ldap` on `apps` to also show `Subordinate(s) upgrade plan`

Reply:
done